<?xml version="1.0"?>
<launch>

  <arg name="node"		default="image_projector"/>
  <arg name="camera_name"	default="live_camera"/>
  <arg name="camera_frame"	default="$(arg camera_name)_optical_frame"/>
  <arg name="cam_param_file"
       default="$(dirname)/../config/CameraParams/Logicool19201080S.txt"/>
  <arg name="mk_param_file"
       default="$(dirname)/../config/MarkerParams/marker94Para.yml"/>
  <arg name="manager"		default="mgr"/>
  <arg name="manager_threads"	default="4"/>
  <arg name="vis"		default="true"/>

  <arg name="image_topic"	value="$(arg camera_name)/image_raw"/>
  <node unless="$(eval manager == '')"
	name="$(arg manager)"
	pkg ="nodelet" type="nodelet" args="manager" output="screen">
    <param name="num_worker_threads"	value="$(arg manager_threads)"/>
  </node>

  <!-- Launch camera -->
  <arg name="video_device"	default="/dev/video8"/>
  <arg name="image_width"	default="1920"/>
  <arg name="image_height"	default="1080"/>
  <arg name="pixel_format"	default="yuyv"/>
  <arg name="fps"		default="5"/>
  <node name="$(arg camera_name)"
  	pkg="usb_cam" type="usb_cam_node" output="screen">
    <param name="video_device"		value="$(arg video_device)"/>
    <param name="image_width"		value="$(arg image_width)"/>
    <param name="image_height"		value="$(arg image_height)"/>
    <param name="pixle_format"		value="$(arg pixel_format)"/>
    <param name="framerate"		value="$(arg fps)"/>
    <param name="camera_frame_id"	value="$(arg camera_frame)"/>
  </node>

  <!-- Launch aist_lenti_mark or image_projector -->
  <include file="$(dirname)/run.launch">
    <arg name="node"			value="$(arg node)"/>
    <arg name="cam_param_file"		value="$(arg cam_param_file)"/>
    <arg name="mk_param_file"		value="$(arg mk_param_file)"/>
    <arg name="image_topic"		value="$(arg image_topic)"/>
    <arg name="manager"			value="$(arg manager)"/>
  </include>

  <node if="$(arg vis)"
	name="$(anon rviz)" pkg="rviz" type="rviz" output="screen"
  	args="-d $(dirname)/$(arg camera_name).rviz"/>

</launch>
